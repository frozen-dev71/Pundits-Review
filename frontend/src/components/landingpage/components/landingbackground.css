/*
// ======================================
// This file contains the CSS to style the background image used on the landing page 
// ======================================
*/


.landing-background-image {
	background-image: -webkit-linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url("https://images.pexels.com/photos/5236674/pexels-photo-5236674.png?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260");
	background-image: -o-linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url("https://images.pexels.com/photos/5236674/pexels-photo-5236674.png?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260");
	background-image: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url("https://images.pexels.com/photos/5236674/pexels-photo-5236674.png?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260");

  /* Set a specific height */
  	height: 100vh;

  /* Position and center the image to scale nicely on all screens */
	background-position: left;
	background-repeat: no-repeat;
	background-size: cover;
	position: relative;
	z-index: -1;
}