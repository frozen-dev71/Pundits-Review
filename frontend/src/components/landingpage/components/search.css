/*
// ======================================
// CSS for the search bar
// ======================================
*/

.search-bar-wrapper {
    border-radius: 1em;
    box-shadow: #909090 1px 1px 10px;
    opacity: .85;
    z-index: 80;
}

.search-box {
	width: 65vw;
    padding: .85em 1em;
    border: none;
    background: none;
    letter-spacing: .05em;
    font-size: .85em;
}

.search-button {
	padding: 0em 1em;
    border: none;
    background: none;
    font-size: 1.25em;
    opacity: .5;
    cursor: pointer;
    height: 100%;
    max-height: 1.75em;
}

.search-button:focus, .search-box:focus {
	outline: none;
}

.fa-search {
    color: black;
}


/*
// ======================================
// MEDIA QUERIES
// ======================================
*/

@media (min-width: 800px) {

    .search-box {
        font-size: .95em;
        width: 50vw;
    }
}



@media (min-width: 1000px) {

    .search-box {

        width: 40vw !important;
    }
}


@media (min-width: 1200px) {

    .search-box {

        width: 35vw !important ;
    }
}
