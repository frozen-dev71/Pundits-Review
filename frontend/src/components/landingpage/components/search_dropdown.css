/*
// ======================================
// CSS for live search dropdown list
// ======================================
*/


.search-dropdown-wrapper {

	max-height: 175px;
	overflow: scroll;
	display: flex;
	flex-direction: column;
	border-radius: 0 0 1em 1em;

}

.inner-dropdown-container {
	cursor: pointer;
	color: black;
}

.inner-dropdown-container:hover {
	background-color: lightgrey !important;
}

.search-dropdown-text {
	margin: 1em 1em;
	font-size: .8em;
}




@media (min-width: 800px) {

    .search-dropdown-text  {
        font-size: .9em;
    }
}