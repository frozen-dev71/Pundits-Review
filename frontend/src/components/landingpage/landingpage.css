/*
// ======================================
// This file contains the top level CSS for the landing page
// ======================================
*/


.landing-logo-text {
	font-size: 7.5em;

}

.landing-site-name-text {
	margin-bottom: 1em;
	font-size: 1.75em;
	letter-spacing: .15em;
}


.landing-page-text-inner-wrapper {
	position: absolute;
    display: flex;
    flex-direction: column;
    top: 12.5vh;
    height: 52.5%;
    width: 100vw;
    justify-content: center;
    align-items: center;
}

.landing-description-text {
	text-align: center;
    letter-spacing: .1em;
    font-size: .725em;
    width: 70%;
}


.landing-page-search-inner-wrapper {
	position: absolute;
    display: flex;
    flex-direction: column;
    bottom: 0;
    width: 95%;
    height: 35%;
    justify-content: flex-start;
    align-items: center;

}



/*
// ======================================
// MEDIA QUERIES - HEIGHT
// ======================================
*/

@media (max-height: 550px) {


    .landing-description-text {
        display: none;
    }

    .landing-site-name-text {
        display: none;
    }

    .landing-page-text-inner-wrapper {
        height: 30%;
    }

    .landing-page-search-inner-wrapper {
        height: 50%;
    }
}


/*
// ======================================
// MEDIA QUERIES - WIDTH
// ======================================
*/

@media (min-width: 500px) {

    .landing-logo-text {
        font-size: 8.5em;
    }

    .landing-site-name-text {
        font-size: 2em;
    }

    .landing-description-text {
        font-size: .825em;
    }
}


@media (min-width: 800px) {

    .search-box {
        width: 50vw;
    }

    .landing-logo-text {
        font-size: 9.5em;
    }


    .landing-description-text {
        font-size: .9em;
    }
}



@media (min-width: 1000px) {

    .landing-description-text {
        width: 50%;
    }

}


@media (min-width: 1200px) {

    .landing-logo-text {
        font-size: 10.5em;
    }

    .landing-site-name-text {
        font-size: 2.25em;
    }

    .landing-description-text {
        font-size: 1em;

    }
}



@media (min-width: 1500px) {

    .landing-logo-text {
        font-size: 11.5em;
    }

    .landing-site-name-text {
        font-size: 2.5em;
    }

    .landing-description-text {
        font-size: 1.15em;
        width: 40%;
    }
}