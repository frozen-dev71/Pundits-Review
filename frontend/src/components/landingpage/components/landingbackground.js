// ======================================
// This file contains the image component that is used to cover the background of the landing page
// Used within landingpage.js
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// Local JS Imports 


// Local CSS Imports
import './landingbackground.css';



// ===============
// LandingBackground Component 
// ===============

export class LandingBackground extends Component {

	render() {

		let result = (

				<div className="landing-background-image">

				</div>

			);


		return result;
	}
}