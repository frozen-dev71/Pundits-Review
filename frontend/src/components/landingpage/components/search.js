// ======================================
// Search bar component allowing users to enter a player/club search query
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { Link, withRouter } from 'react-router-dom';
import { connect } from "react-redux";
import axios from 'axios';

// Local JS Imports 
import { defineSearchTerm, searchByPlayer, searchByClub, clearSearchResults, setThisWeekEveryoneToggleStatus } from "../../actions/actions";
import { SearchDropdownWrapper } from './search_dropdown';
import { makeSearch } from './cancel_requests';  

// Local CSS Imports
import './search.css';

// ===============
// Search Component 
// ===============

export class Search extends Component {

	constructor(props) {
    super(props);
    this.state = {
    		searchTerm: "",
    		loading: false,
    		liveSearchResultsPlayers: null,
    		liveSearchResultsClubs: null,
    	};

    this._handleChange = this._handleChange.bind(this);
    this._handleClick = this._handleClick.bind(this);
    this.liveSearch = this.liveSearch.bind(this);

  	}

  	// Keep search term inside search input box on results page
  	componentDidMount = () => {

  		if (this.props.searchTerms.length && this.props.history.location.pathname === "/results") {

  			// Set the state to the previous search's value
  			this.setState({ searchTerm: this.props.searchTerms[this.props.searchTerms.length -1]});

  		}

  	}


  	// Predictive search ---> updates search options when search term changes
  	liveSearch = async val => {

  		// Set the state to loading
  		this.setState({ loading: true });

  		// Make the async request
  		const resPlayers = await makeSearch(
	      `/api/plainplayers/?search=${val}`
	    )

	    const resClubs = await makeSearch(
	      `/api/plainclubs/?search=${val}`
	    )

  		// Wait for the search to complete
  		const liveSearchResultsPlayers = await resPlayers;
  		const liveSearchResultsClubs = await resClubs;

  		// Set the state to not loading
	    this.setState({ 
	    	liveSearchResultsPlayers,
			liveSearchResultsClubs,
	    	loading: false 
	    });
  	}

  	_handleKeyDown(e) {

  		// Trigger Search on Enter
  		if (e.key == "Enter") {
  			this._handleClick();
  			this.props.history.push('/results');
  		}
  	}


  	_handleChange = async searchTerm => {

  		// Ensure that dropdown menu appears 
  		let dropdown = document.querySelector('.search-dropdown-wrapper');

  		if (dropdown) {
  			dropdown.style.display = "flex";
  		}

		// Trigger a live search with the search term
		this.liveSearch(searchTerm);

		// Update the state with the search term
		this.setState({searchTerm: searchTerm})

  		
  	}

	_handleClick() {
		
		// Set redux state 	

		// Clear the previous search results
		this.props.clearSearchResults();

		// Define the search term
		this.props.defineSearchTerm(this.state.searchTerm);

		// Search By Player
		this.props.searchByPlayer(this.state.searchTerm);

		// Search By Club
		this.props.searchByClub(this.state.searchTerm);

		// Set This Week Everyone toggle to this week
		this.props.setThisWeekEveryoneToggleStatus("THIS WEEK")


		// Ensure that dropdown menu disappears 
  		let dropdown = document.querySelector('.search-dropdown-wrapper');

  		if (dropdown) {
  			dropdown.style.display = "none";
  		}

	}


	get renderSearchResults() {

		let searchDropdown;

		if (this.state.searchTerm !== "") {

			// Set the search dropdown to display results
			searchDropdown = <SearchDropdownWrapper playerResults={this.state.liveSearchResultsPlayers} clubResults={this.state.liveSearchResultsClubs}/>
		} else {
			// Set the search dropdown to display an empty span
			searchDropdown = <span></span>;
		}

		return searchDropdown;
	}

	render() {

		let result = (

				<div className="search-bar-wrapper background-light">

			      	<input value={this.state.searchTerm} aria-label="Search for a player or a club" aria-required="true" type="text" name="Search input to search for Premier League players or clubs." id="search-bar-input" autoComplete="off" className="search-box body-text-light" placeholder="Search for a player or a club ..." onChange={(e) => this._handleChange(e.target.value)} onKeyDown={(e) => this._handleKeyDown(e)}/>
			      	<Link className="react-link" to="/results">
			      		<button type="submit" id="search-button" className="search-button" onClick={this._handleClick} >
			        		<i className="fa fa-search"></i>
			        	</button>
			     	</Link>
					{this.renderSearchResults}
				</div>
			);

		return result;
	}
}


// ===============
// Connect redux state container to search component
// ===============

const mapStateToProps = state => {
  return { 
	  	searchTerms: state.searchTerms,
	  	searchDataPlayers: state.searchDataPlayers,
	  	searchDataClubs: state.searchDataClubs
  	};
};


// Export connected component
export const SearchBar = connect(mapStateToProps, { defineSearchTerm, searchByPlayer, searchByClub, clearSearchResults, setThisWeekEveryoneToggleStatus })(withRouter(Search));

