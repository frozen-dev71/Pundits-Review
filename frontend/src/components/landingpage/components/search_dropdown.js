// ======================================
// Search Dropdown component - Rendered when Live Search is triggered
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { Link, withRouter } from 'react-router-dom';
import { connect } from "react-redux";

// Local JS Imports
import { defineSearchTerm, searchByPlayer, searchByClub, clearSearchResults, setThisWeekEveryoneToggleStatus } from "../../actions/actions";


// CSS Imports
import './search_dropdown.css';

// ===========
// Search Dropdown Component
// ===========


export class SearchDropdown extends Component {

	_handleClick(e) {

		let targetClicked = e.target.innerText

		// Set redux state 	

		// Clear the previous search results
		this.props.clearSearchResults();

		// Define the search term
		this.props.defineSearchTerm(targetClicked);

		// Search By Player
		this.props.searchByPlayer(targetClicked)

		// Search By Club
		this.props.searchByClub(targetClicked)

		// Reset this week everyone toggle
		this.props.setThisWeekEveryoneToggleStatus("THIS WEEK")
		
		// Hide the dropdown menu & reset input menu value to empty string
		document.querySelector('.search-dropdown-wrapper').style.display = "none";
		document.querySelector('#search-bar-input').value = ""

	}



	render() {

		let result = (

				<div className="search-dropdown-wrapper background-light">

					{this.props.playerResults ? this.props.playerResults.data.map((searchResult, index) => {

						let innerDropdownContainer = (

								<Link key={index} to="/results" className="inner-dropdown-container react-link" onClick={(e) => this._handleClick(e)}>
									<p className="search-dropdown-text body-text-light">{searchResult.player}</p>
								</Link>

							) 

						return innerDropdownContainer;

					}): <span></span>}

					{this.props.clubResults ? this.props.clubResults.data.map((searchResult, index) => {

						let innerDropdownContainer = (

								<Link key={index} to="/results" className="inner-dropdown-container react-link" onClick={(e) => this._handleClick(e)}>
									<p className="search-dropdown-text body-text-light">{searchResult.club}</p>
								</Link>

							) 

						return innerDropdownContainer;

					}): <span></span>}


				</div>

			);

		return result; 

	}
}



// ===============
// Connect redux state container to search dropdown component
// ===============

const mapStateToProps = state => {
  return { 
	  	searchTerms: state.searchTerms,
	  	searchDataPlayers: state.searchDataPlayers,
	  	searchDataClubs: state.searchDataClubs
  	};
};


// Export connected component
export const SearchDropdownWrapper = connect(mapStateToProps, { defineSearchTerm, searchByPlayer, searchByClub, clearSearchResults, setThisWeekEveryoneToggleStatus })(SearchDropdown);