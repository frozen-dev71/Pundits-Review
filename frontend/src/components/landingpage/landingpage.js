// ======================================
// This file contains the top level LANDING PAGE component that will wrap other elements on the landing page
// All other LANDING PAGE components will be directed through this component
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// Local JS Imports 
import { LandingBackground } from './components/landingbackground';
import { SearchBar } from './components/search';

// Local CSS Imports
import './landingpage.css';

// ===============
// LandingPage Component 
// ===============


export class LandingPage extends Component {

	render() {

		let result = (

			<div className="landing-page-wrapper">

				<LandingBackground />

				<div className="landing-page-text-inner-wrapper">

					<p className="landing-logo-text heading-text white-text">PR</p>

					<p className="landing-site-name-text heading-text white-text">Pundits Review</p>

					<p className="landing-description-text body-text-light white-text side-margin">The scoring system that shows you how Premier League players and teams were reviewed by the media</p>

				</div>

				<div className="landing-page-search-inner-wrapper side-margin">
					<SearchBar />
				</div>

			</div>

			);

		return result;
	}
}