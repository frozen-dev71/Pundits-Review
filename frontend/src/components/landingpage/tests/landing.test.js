// ======================================
// This file tests for the components listed in the LANDINGPAGE sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that LandingPage renders correctly
// 2. Check that LandingBackground renders correctly
// 3. Check that Search renders correctly
// 4. Check that SearchDropdown renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

// Local JS Imports
import { LandingPage } from '.././landingpage';
import { LandingBackground } from '.././components/landingbackground';
import { Search } from '.././components/search';
import { SearchDropdown } from '.././components/search_dropdown';

import store from '../../store/store';

// Configurations
Enzyme.configure({ adapter: new Adapter() })

// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" & 'Warning: React.createElement: type is invalid -- expected a string (for built-in components)' - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('Landing Page Component Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server') || msg.includes('Warning: React.createElement: type is invalid -- expected a string (for built-in components)')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });


// End of referenced code

  	// ===========
	// TEST 1. Check that LandingPage renders correctly
	// ===========

	describe('LandingPage', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><LandingPage/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 2. Check that LandingBackground renders correctly
	// ===========

	describe('LandingBackground', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><LandingBackground/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 3. Check that Search renders correctly
	// ===========

	describe('Search', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><Search/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that SearchDropdown renders correctly
	// ===========

	describe('SearchDropdown', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><SearchDropdown/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


});
