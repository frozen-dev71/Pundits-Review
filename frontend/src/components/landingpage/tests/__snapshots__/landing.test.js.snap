// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Component Tests LandingBackground should render correctly 1`] = `
<div
  class="landing-background-image"
/>
`;

exports[`Landing Page Component Tests LandingPage should render correctly 1`] = `
<div
  class="landing-page-wrapper"
>
  <div
    class="landing-background-image"
  />
  <div
    class="landing-page-text-inner-wrapper"
  >
    <p
      class="landing-logo-text heading-text white-text"
    >
      PR
    </p>
    <p
      class="landing-site-name-text heading-text white-text"
    >
      Pundits Review
    </p>
    <p
      class="landing-description-text body-text-light white-text side-margin"
    >
      The scoring system that shows you how Premier League players and teams were reviewed by the media
    </p>
  </div>
  <div
    class="landing-page-search-inner-wrapper side-margin"
  >
    <div
      class="search-bar-wrapper background-light"
    >
      <input
        aria-label="Search for a player or a club"
        aria-required="true"
        autocomplete="off"
        class="search-box body-text-light"
        id="search-bar-input"
        name="Search input to search for Premier League players or clubs."
        placeholder="Search for a player or a club ..."
        type="text"
        value=""
      />
      <a
        class="react-link"
        href="/results"
      >
        <button
          class="search-button"
          id="search-button"
          type="submit"
        >
          <i
            class="fa fa-search"
          />
        </button>
      </a>
      <span />
    </div>
  </div>
</div>
`;

exports[`Landing Page Component Tests Search should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Search />
  </Provider>
</Router>
`;

exports[`Landing Page Component Tests SearchDropdown should render correctly 1`] = `
<div
  class="search-dropdown-wrapper background-light"
>
  <span />
  <span />
</div>
`;
